### PR TITLE
upstream: not trying to do HTTP/3 where not configured

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: new unit tests
Docs Changes: updated API docs
Release Notes: n/a

### CODE REVIEW

File: test/common/upstream/upstream_impl_test.cc [hunk 1]
```diff
@@ -3395,6 +3396,41 @@ TEST_F(ClusterInfoImplTest, Http3BadConfig) {
   EXPECT_THROW_WITH_REGEX(makeCluster(yaml), EnvoyException,
                           "HTTP3 configured but not enabled in the build.");
 }
+TEST_F(ClusterInfoImplTest, UseDownstreamHttpProtocolWithoutDowngrade) {
```
Comment (asraa) [inline test/common/upstream/upstream_impl_test.cc:3399]: I think this may need to go above `#else` at least?

Comment (alyssawilk) [reply test/common/upstream/upstream_impl_test.cc]: Indeed, nice catch :-)

**Comment by alyssawilk** [comment]
CI sorted. Lizan PTAL!

**Review by lizan — APPROVED** [review]

**Pull request merged**